presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -251,8 +251,16 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> contex
         @Override
         public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Expression> context)
         {
-            checkState(!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol()), "predicate depends on marker symbol");
-            return context.defaultRewrite(node, context.get());
+            Set<Symbol> pushDownableSymbols = ImmutableSet.copyOf(node.getDistinctSymbols());
```

Comment:
add test to `TestPredicatePushdown`?

Reply:
It is not much possible the only query which is using this is the correlated scalar subquery, it would be awkward to test that there. 

Maybe once this PR got merged I could add such test at the top.